### DISCUSSION TITLE
v1.91.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#5723] [Announcements]
# v1.91.0

Welcome to the release `v1.91.0` of Immich! This release is _packed_ with changes. Some of the highlights include:
- A list of breaking changes/action required steps
- More details about the removal of the Typesense container and switch to pgvecto.rs, including some FAQs
- Notable fix: Backing up assets from iCloud no longer hangs the process

> [!IMPORTANT]
> # Action Required
> 
> 1. `docker-compose.yml` updates related to dropping Typesense
> 2. Reupload certain iOS Live Photos
> 3. Changes to the `LOG_LEVEL` environment variable

## 1. `docker-compose.yml` updates

We are _removing the Typesense container_ and _changing the database image_. Below are the changes that must be made in your `docker-compose.yml` file.

```diff
  immich-server:
  [...]
    depends_on:
      - redis
      - database
-     - typesense
    restart: always

  immich-microservices:
  [...]
    depends_on:
      - redis
      - database
-     - typesense
    restart: always

-  typesense:
-    container_name: immich_typesense
-    image: typesense/typesense:0.24.1@sha256:9bcff2b829f12074426ca044b56160ca9d777a0c488303469143dd9f8259d4dd
-    environment:
-      - TYPESENSE_API_KEY=${TYPESENSE_API_KEY}
-      - TYPESENSE_DATA_DIR=/data
-      # remove this to get debug messages
-      - GLOG_minloglevel=1
-    volumes:
-      - tsdata:/data
-    restart: always

[...]

  database:
    container_name: immich_postgres
-   image: postgres:14-alpine@sha256:6a0e35296341e676fe6bd8d236c72afffe2dfe3d7eb9c2405c0f3fc04500cd07
+   image: tensorchord/pgvecto-rs:pg14-v0.1.11
    env_file:
      - .env
    environment:

volumes:
  pgdata:
  model-cache:
- tsdata:
```
> 
>  [!NOTE]
> **Note:** If you are running your database with a non-superuser role for Immich, you must enable the pgvecto.rs extension _manually_. You can do this by connecting to the `immich` database as a superuser and running:
> ```sql
> CREATE EXTENSION vectors;
> ```

> [!NOTE]
> See below for more details about this change, including frequently asked questions.

## 2. Reupload certain iOS Live Photos

iOS Live Photos **uploaded after v1.89.0** that are **not linked** need to be deleted and re-uploaded from the mobile app.

This is a one-time action, and future live photos uploaded from the mobile app will be properly linked together.

## 3. Changes to the `LOG_LEVEL` environment variable
The `LOG_LEVEL` value of `simple` has been removed. The equivalent value is `log`. If you were using the value `simple`, the server container will not start until this is updated.

### I kept Typesense disabled because my CPU doesn’t support AVX. Does this change mean I can’t use Immich anymore?

The new vector search extension we’re using, pgvecto.rs, does not require AVX to function as it does a runtime check on the SIMD instructions your CPU supports.

### Does it work with Raspberry Pi?

It is tested to work with Raspberry Pi 5, and should work with Raspberry Pi 4 as well.

### Do I need to do a backup and restore for my Postgres database?

While it is recommended to regularly backup your database, this change can be done in-place just by changing the image. You do not need to do a backup and restore unless you run into a specific issue that requires it. 

### I have a common Postgres instance that I share with other services. What do I need to do with this change?

The Postgres image we use is regular Postgres, just with the pgvecto.rs extension. If your Postgres instance is a regular instance without any third-party extensions, then you can simply switch out the image for that instance with the `tensorchord/pgvecto-rs:pg14-v0.1.11` image (changing `pg14` to the major version you use - 14, 15 or 16). This is a drop-in replacement that will work without a backup and restore.

If your instance has third-party extensions, then you will need to make a docker image that installs the pgvecto.rs extension in addition to the other extensions based on their [installation instructions](https://github.com/tensorchord/pgvecto.rs/blob/main/docs/installation.md#install-from-release). An example of this for the CloudNativePG Kubernetes operator can be found [here](https://github.com/bo0tzz/cnpgvecto.rs).

If you use a bare-metal instance and have a Debian-based server, then you may instead follow their [installation instructions](https://github.com/tensorchord/pgvecto.rs/blob/main/docs/installation.md#install-from-release) and install the pgvecto.rs Debian [package](https://github.com/tensorchord/pgvecto.rs/releases/tag/v0.1.11) directly.

### While trying to backup and restore my Postgres instance, I got the error `type “earth” does not exist`. What do I do?

This is a bug with Postgres’ earthdistance extension. You can fix the issue by following the instructions [here](https://github.com/immich-app/immich/issues/5630#issuecomment-1851015572).

## Search Background

It turns out that synchronizing data between Postgres and Typesense is quite complicated. It accounts for at least an additional 1000 lines of code and, while very feature-filled, has become a burden to maintain. Also, there have been some _serious_ memory and performance issues with Typesense, especially with large photo collections. It is lightning fast, but at the expense of large indexes that must be read from disk into memory on each restart. There have been reports of this process taking up to _30 minutes_!

We effectively remove an entire class of bugs and issues by removing the need to synchronize data. Furthermore, search queries can more easily be combined with the existing database schema, like libraries, partner sharing, albums, etc. After merging #3605, we hope to be able to make significant progress around the search implementation and feature set.

In short, some of the benefits of this change include the following:
- One less container to run, deploy, and manage
- Better memory/system resource usage
- Decreased memory usage
- Faster Encode CLIP and Recognize Faces jobs
- Faster startup time
- Significantly less code to maintain, test, debug
- Improved developer workflow when adding search-related functionality
- Ability to integrate smart search into end-to-end testing for improved testing coverage and stability
- Ability to integrate CLIP search with other (metadata) criteria in the future
- Ability to add customizable index settings for even lower memory usage and higher performance, such as on-disk indices and quantization

- - - -

And as always, bugs are fixed, and many other improvements also come with this release.

Please consider supporting the project.

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich via the following channels.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 1FvEp6P6NM8EZEkpGUFAN2LqJ1gxusNxZX

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

### Breaking Changes 🛠
* feat(web,server)!: runtime log level by @jrasm91 in https://github.com/immich-app/immich/pull/5672
* feat(server): use pgvecto.rs by @jrasm91 in https://github.com/immich-app/immich/pull/3605

### Server
* feat(server): use postgres-adapter for websockets by @jrasm91 in https://github.com/immich-app/immich/pull/5569
* feat(server,ml): add new models by @mertalev in https://github.com/immich-app/immich/pull/5710
* feat(server): server-side events by @jrasm91 in https://github.com/immich-app/immich/pull/5669
* fix(server): disable classification by default by @mertalev in https://github.com/immich-app/immich/pull/5708
* fix(server): select asset face columns explicitly by @mertalev in https://github.com/immich-app/immich/pull/5564
* fix(server): connection aborted logging by @jrasm91 in https://github.com/immich-app/immich/pull/5595
* fix(server): don't associate assets with Null Island by @sushain97 in https://github.com/immich-app/immich/pull/5623
* fix(server): in dev mode by @jrasm91 in https://github.com/immich-app/immich/pull/5670
* chore(server): simplify search face query and better clustering by @alextran1502 in https://github.com/immich-app/immich/pull/5573
* chore(server): faster shutdown by @jrasm91 in https://github.com/immich-app/immich/pull/5577
* chore(server): fix ssr in dev by @jrasm91 in https://github.com/immich-app/immich/pull/5637
* refactor(server): auth dto by @jrasm91 in https://github.com/immich-app/immich/pull/5593
* refactor(server): immich file responses by @jrasm91 in https://github.com/immich-app/immich/pull/5641

### Mobile
* fix(mobile): handle livePhotos using originFileWithSubType by @shenlong-tanwen in https://github.com/immich-app/immich/pull/5602
* fix(mobile): mobile album sort not persisting by @shenlong-tanwen in https://github.com/immich-app/immich/pull/5584
* fix(mobile): memory lane not displayed in mobile app by @shenlong-tanwen in https://github.com/immich-app/immich/pull/5587
* fix(mobile): live / motion photo download by @shenlong-tanwen in https://github.com/immich-app/immich/pull/5607
* fix(mobile): Revert - upload motion and live part of LivePhotos together by @alextran1502 in https://github.com/immich-app/immich/pull/5601
* fix(mobile): better error message by @alextran1502 in https://github.com/immich-app/immich/pull/5653
* fix(mobile): Fix upload hang on iOS when deleting stale files by @alextran1502 in https://github.com/immich-app/immich/pull/5658
* fix(mobile): simplify state management in backup selection page by @alextran1502 in https://github.com/immich-app/immich/pull/5655
* chore(mobile): move mocktail to dev dep by @shenlong-tanwen in https://github.com/immich-app/immich/pull/5666
* chore(mobile): Put delete button before metadata editing by @lukaszwawrzyk in https://github.com/immich-app/immich/pull/5633

### Web
* feat(web): use `<a>` tag for albums in list view by @martabal in https://github.com/immich-app/immich/pull/5645
* feat(web): increase map max zoom by @waclaw66 in https://github.com/immich-app/immich/pull/5693
* feat(web): lazy loading on album/sharing/search by @flyotlin in https://github.com/immich-app/immich/pull/5696
* feat(web): add types to dispatcher by @martabal in https://github.com/immich-app/immich/pull/5700
* feat(web): re-add open graph tags for public share links by @jrasm91 in https://github.com/immich-app/immich/pull/5635
* feat(web): automatically update user info by @martabal in https://github.com/immich-app/immich/pull/5647
* fix(web): user management responsive design by @martabal in https://github.com/immich-app/immich/pull/5698
* fix(web): log out by @martabal in https://github.com/immich-app/immich/pull/5706
* fix(web): remove always on delay badge by @alextran1502 in https://github.com/immich-app/immich/pull/5574
* fix(web): no icon with firefox by @martabal in https://github.com/immich-app/immich/pull/5679
* fix(web): warning when building web by @martabal in https://github.com/immich-app/immich/pull/5680
* fix(web): prevent loop on search by @martabal in https://github.com/immich-app/immich/pull/5664
* chore(web): shared page redirect to host by @JW-CH in https://github.com/immich-app/immich/pull/5678

### Machine Learning
* fix(ml): more graceful shutdown by @mertalev in https://github.com/immich-app/immich/pull/5689

### Documentation
* docs: clean-up old references by @jrasm91 in https://github.com/immich-app/immich/pull/5697
* docs: update arch diagram by @jrasm91 in https://github.com/immich-app/immich/pull/5662
* docs: Add tip showing common error messages by @jonhnet in https://github.com/immich-app/immich/pull/5592
* docs: add sponsor link by @jrasm91 in https://github.com/immich-app/immich/pull/5597
* docs: remove typesense from faq by @mertalev in https://github.com/immich-app/immich/pull/5600
* docs: update backup-and-restore.md by @pjsxw in https://github.com/immich-app/immich/pull/5616
* docs: FAQ for thumbnail jobs by @bo0tzz in https://github.com/immich-app/immich/pull/5661
* docs: recap 2023 by @alextran1502 in https://github.com/immich-app/immich/pull/5665
* docs: remove citiesFileOverride from config-file.md; fixes #5682 by @SlavikCA in https://github.com/immich-app/immich/pull/5683

### Dependency updates
* fix(deps): update server by @renovate in https://github.com/immich-app/immich/pull/5506
* chore(deps): update base-image to v20231214 (major) by @renovate in https://github.com/immich-app/immich/pull/5705
* chore(deps): update mambaorg/micromamba:bookworm-slim docker digest to 5ea70d2 by @renovate in https://github.com/immich-app/immich/pull/5707
* chore(deps): update dependency sql-formatter to v15 by @renovate in https://github.com/immich-app/immich/pull/5709
* fix(deps): bump sharp by @jrasm91 in https://github.com/immich-app/immich/pull/5509
* chore(deps): remove unused cron types by @jrasm91 in https://github.com/immich-app/immich/pull/5563
* chore(deps): pin dependencies by @renovate in https://github.com/immich-app/immich/pull/5567
* chore(deps): pin tensorchord/pgvecto-rs docker tag to 0335a1a by @renovate in https://github.com/immich-app/immich/pull/5570
* chore(deps): update python:3.11-slim-bookworm docker digest to cfd7ed5 by @renovate in https://github.com/immich-app/immich/pull/5572
* chore(deps): update redis:6.2-alpine docker digest to b6124ab by @renovate in https://github.com/immich-app/immich/pull/5599
* chore(deps): pin tensorchord/pgvecto-rs docker tag to 0335a1a by @renovate in https://github.com/immich-app/immich/pull/5632
* chore(deps): bump github/codeql-action from 2 to 3 by @dependabot in https://github.com/immich-app/immich/pull/5694

### Other changes
* ci: Automatic categories in generated release notes by @bo0tzz in https://github.com/immich-app/immich/pull/5684
* ci: Catchall category for release notes by @bo0tzz in https://github.com/immich-app/immich/pull/5701

## New Contributors
* @jonhnet made their first contribution in https://github.com/immich-app/immich/pull/5592
* @pjsxw made their first contribution in https://github.com/immich-app/immich/pull/5616
* @lukaszwawrzyk made their first contribution in https://github.com/immich-app/immich/pull/5633
* @JW-CH made their first contribution in https://github.com/immich-app/immich/pull/5678
* @SlavikCA made their first contribution in https://github.com/immich-app/immich/pull/5683

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.90.2...v1.91.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.91.0'>v1.91.0</a>.</em>

**Comment by FelixBuehler** [comment]
Anyone else getting a "Failed to search (500 - Internal Server Error)" when trying to open a location in the Explore > places list?

**Reply by andrewdunndev** [reply]
I was just discussing this in [discord with @jrasm91](https://discord.com/channels/979116623879368755/994044917355663450/1185254291141832814): 

> I think we forgot to include a note about some of the changes to the explore page in the release notes for this one. TLDR, major changes to all search related stuff and we had to remove some existing functionality, which we'll probably add back (with a new implementation) in the future.

**Reply by ipnetxyz** [reply]
I have the same @FelixBuehler

**Reply by vitobotta** [reply]
Same here.

**Comment by davidXire** [comment]
After update, it's impossible to login anymore... 
I got "Incorrect email or password"

When I check the database with SELECT * FROM "users";, my account are here 
When I tryed to reset my password with "immich-admin reset-admin-password" -> no effect, same error

**Reply by alextran1502** [reply]
Is this on the web or mobile app?

**Reply by davidXire** [reply]
Both...
Logs from the server : 
immich_server            | [Nest] 7  - 12/15/2023, 4:33:56 PM    WARN [AuthService] Failed login attempt for user ******@****.com from ip address xxxx

**Reply by davidXire** [reply]
Résolution here : https://discord.com/channels/979116623879368755/1185315110395322528

**Reply by sogg** [reply]
Can someone post the solution here from Discord ,please ？

**Reply by Chaphasilor** [reply]
In the end it boiled down to manually updating the email using SQL. The below query was used for that specific person, but you should definitely modify it to fit your setup before executive it. For example, the WHERE-clause will update the email of all admin users, and if there is more than one admin this will probably cause new issues.  

```sql
UPDATE "users"
SET "email" = 'email@email.com'
WHERE "users"."isAdmin" IS TRUE
```

After updating the email, try login in with the new email (`email@email.com` in this case) and the old password, and then update the email in account settings.

**Reply by sogg** [reply]
thanks!

**Reply by deiop** [reply]
same issue her, all users work. Except admin user. 
Updating the email worked. Thank you

**Comment by ipnetxyz** [comment]
Just did the upgrade and it worked fine (except that part listed above by @FelixBuehler which seems I have).

A short note to fellow users here, just follow the instructions and it will work fine. 

The only part that was a bit sketchy for me was the instruction for the creation of "vectors" extension. I was wondering if it will be created automatically or I have to do it.

What I did was checking the before upgrade and after upgrade the existing extensions.
Here the steps (applies to regular unmodified installation of immich, so if you customized things it may not apply to you)

1) Connect to your DB before upgrade to 1.91.0
```
docker exec -it immich\_postgres psql immich postgres
```
and check what extensions you have there using the command
```
SHOW \dx
```
A sample output from my installation:
```
immich-# SHOW \dx
                                    List of installed extensions
     Name      | Version |   Schema   |                         Description
---------------+---------+------------+--------------------------------------------------------------
 cube          | 1.5     | public     | data type for multidimensional cubes
 earthdistance | 1.1     | public     | calculate great-circle distances on the surface of the Earth
 plpgsql       | 1.0     | pg_catalog | PL/pgSQL procedural language
 uuid-ossp     | 1.1     | public     | generate universally unique identifiers (UUIDs)
(4 rows)
```

2) Upgrade as per instructions

3) Check again the installed extensions
```
docker exec -it immich\_postgres psql immich postgres
```
```
immich-# SHOW \dx
                                    List of installed extensions
     Name      | Version |   Schema   |                         Description
---------------+---------+------------+--------------------------------------------------------------
 cube          | 1.5     | public     | data type for multidimensional cubes
 earthdistance | 1.1     | public     | calculate great-circle distances on the surface of the Earth
 plpgsql       | 1.0     | pg_catalog | PL/pgSQL procedural language
 uuid-ossp     | 1.1     | public     | generate universally unique identifiers (UUIDs)
(4 rows)
```
For me the extension "vectors" was not present, thus I had to install it manually as per instructions.

4) Install the extension and check again:
```
docker exec -it immich\_postgres psql immich postgres
```

```
immich=# CREATE EXTENSION vectors;
CREATE EXTENSION
immich=# SHOW \dx
                                                    List of installed extensions
     Name      | Version |   Schema   |                                         Description
---------------+---------+------------+----------------------------------------------------------------------------------------------
 cube          | 1.5     | public     | data type for multidimensional cubes
 earthdistance | 1.1     | public     | calculate great-circle distances on the surface of the Earth
 plpgsql       | 1.0     | pg_catalog | PL/pgSQL procedural language
 uuid-ossp     | 1.1     | public     | generate universally unique identifiers (UUIDs)
 vectors       | 0.1.1   | public     | vectors: Vector database plugin for Postgres, written in Rust, specifically designed for LLM
(5 rows)
```

Now the extension is there.

I hope this helps!

**Reply by eyebinker** [reply]
Figured it out.  I changed the password in the .env and just needed to add --password at then end of the command so it would ask for the DB password


I tried connecting to the DB and and keep getting this error
Ipsql: error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: FATAL:  role "postgres" does not exist

**Reply by alien75** [reply]
That was useful, thanks!

**Comment by waclaw66** [comment]
I can see a lot of these...
```
pro 15 18:42:27 server immich-server[1074140]: Error [ERR_HTTP_HEADERS_SENT]: Cannot remove headers after they are sent to the client
pro 15 18:42:27 server immich-server[1074140]:     at ServerResponse.removeHeader (node:_http_outgoing:782:11)
pro 15 18:42:27 server immich-server[1074140]:     at ServerResponse.send (/opt/immich/server/node_modules/express/lib/response.js:214:10)
pro 15 18:42:27 server immich-server[1074140]:     at ExpressAdapter.reply (/opt/immich/server/node_modules/@nestjs/platform-express/adapters/express-adapter.js:31:29)
pro 15 18:42:27 server immich-server[1074140]:     at RouterResponseController.apply (/opt/immich/server/node_modules/@nestjs/core/router/router-response-controller.js:15:36)
pro 15 18:42:27 server immich-server[1074140]:     at /opt/immich/server/node_modules/@nestjs/core/router/router-execution-context.js:175:48
pro 15 18:42:27 server immich-server[1074140]:     at async /opt/immich/server/node_modules/@nestjs/core/router/router-execution-context.js:47:13
pro 15 18:42:27 server immich-server[1074140]:     at async /opt/immich/server/node_modules/@nestjs/core/router/router-proxy.js:9:17
```
Everything works apart of that.

**Reply by Eirikr70** [reply]
Same here

**Reply by dsm1212** [reply]
I have the same issue. I'm using only external read-only volumes. I'm using swag (nginx proxy) if that has anything to do with this.

**Reply by adambeck7** [reply]
Same, but it works fine if accessed locally, so it's probably something with our proxy configs. I'll try to dig into it tomorrow morning when I have some time.

**Reply by adambeck7** [reply]
@Eirikr70 @dsm1212 I updated to 1.91.2 and that seems to have fixed the issue for me.

**Reply by Eirikr70** [reply]
@adambeck7 You're right. Thanks ! 👍

**Comment by waclaw66** [comment]
Additionally...
```
pro 15 19:05:55 server immich-server[1088951]: [Nest] 1088951  - 15. 12. 2023 19:05:55    WARN [ExpressAdapter] Content-Type doesn't match Reply body, you might need a custom ExceptionFilter for non-JSON responses
```

while accessing `/photos` or `/albums` directly. The response is an empty page.

**Reply by waclaw66** [reply]
Anyone has the same problem while reloading non-root URL (/photos, /sharing, /albums)?

**Reply by alextran1502** [reply]
Hmm I can't reproduce. Try logout and log back in

**Reply by waclaw66** [reply]
Didn't help :(

**Reply by waclaw66** [reply]
Tried to access immich-server directly without reverse nginx reverse proxy, the same result. Verbose log shows nothing. 1.90.2 worked fine.

**Comment by Solarunit** [comment]
Just updated without issues, thanks!
I wish it was possible to disable Search, Map, Sharing features that I don't use at all. I assume it also should improve performance.

**Reply by jrasm91** [reply]
Look in:
- Admin > Settings for Maps and 
- User Settings > Sidebar for sharing.

**Reply by Solarunit** [reply]
thanks! I also found Machine learning settings! And disabled it :)
Nice!

**Comment by JVT038** [comment]
I have been using the `14-bullseye` postgres image instead of the default version 14, due to Unix sockets.

When I upgraded, I encountered the error `pgvecto.rs: pgvecto.rs must be loaded via shared_preload_libraries.`. After some searching, I discovered that I had to go into my postgres data directory and change `postgresql.conf`. 

Uncommenting the line `shared_preload_libraries` and setting it to `vectors.so` (so it's like `shared_preload_libraries = 'vectors.so'`) fixed the issue for me. 

Hope this helps anyone else who has the same problem.

**Reply by hoggatt** [reply]
Why use a single db instead of containerized versions for each service? Unless you're short on resources, seems like a no-brainer from a maintainence perspective.

**Comment by lo97** [comment]
After updating, now "Tag objects" in the administration tab is "Disabled". Why?

**Reply by mertalev** [reply]
This job was causing the ML service to crash on some ARM devices. We're planning on replacing it with smart albums in the future anyway, so it's better to disable it to avoid this issue for now.

**Reply by mertalev** [reply]
But if this job worked for you in a previous version, then you can re-enable it in the admin settings. The change was just to the default setting.

**Reply by lo97** [reply]
Ok, thank you. Mine is not an arm device so i'll try to re-enable it

**Comment by ghost** [comment]
When I click on a place on the explore tab, I get a 500 error

```
immich_server            | [Nest] 7  - 12/15/2023, 10:10:32 PM   ERROR [ErrorInterceptor] Failed to search
immich_server            | [Nest] 7  - 12/15/2023, 10:10:32 PM   ERROR [ErrorInterceptor] Error: Missing query
```

**Reply by alextran1502** [reply]
https://github.com/immich-app/immich/issues/5726#issuecomment-1858245574

**Comment by ghost** [comment]
It also seems like search is broken?  When I typed dog before this change I got 100s of photos, now I get 1.  If I use m:dog, I still get 100s.

**Comment by vitobotta** [comment]
After upgrading search with the `m:` prefix for metadata is no longer working for me. e.g. `m:wedding` used to return stuff about weddings but now no results. Same thing with other searches. Is it expected with this release?

**Reply by mertalev** [reply]
No, metadata search was tested to work before the release so it seems like there's a regression there.

**Reply by vitobotta** [reply]
I see, thanks :)

**Comment by apetersson** [comment]
experience from upgrading from 1.87.0 to 1.91.0 on unraid (with docker-compse stack)
i updated the docker-compose to the 1.91.0 template (plus backup scripts)
then i had to manually stop the 3 containers no longer in use (immich_proxy, immich_typesense, immich_web)

after the upgrade, the swag was no longer pointing at the right container/port, so this change was needed in my 

/mnt/user/appdata/swag/nginx/proxy-confs/immich.subdomain.conf:

```
server {
    listen 443 ssl;
    listen [::]:443 ssl;

    server_name photos.*;

    include /config/nginx/ssl.conf;

    client_max_body_size 0;

    location / {
        include /config/nginx/proxy.conf;
        include /config/nginx/resolver.conf;
# old value        set $upstream_app immich_proxy;
# old value      set $upstream_port 8080;
        set $upstream_app immich_server; # new value
        set $upstream_port 3001; # new value
        set $upstream_proto http;
        proxy_pass $upstream_proto://$upstream_app:$upstream_port;

    }
}

```
so far, everything seems to work

**Reply by alextran1502** [reply]
Try change this `set $upstream_port 3001;` to `set $upstream_port 2283;`

**Reply by mikkelsiggaard** [reply]
@alextran1502
The typesense container isn't stopped automatically after upgrading, so pruning won't affect it. 
I had to do `docker stop [CONTAINER ID]` and then `docker system prune` which removed ~800MB of space for me. 
Is this the recommended method for a vanilla docker install (i.e. non-unraid)?

**Reply by alextran1502** [reply]
there are many ways to remove a container, your method is fine

**Comment by kushfest** [comment]
I've upgraded and so far it seems all was successful. I'm unclar how I can tell which iOS Live Photos I need to erase and re-upload. Am I just looking for anything that was uploaded since 1.89.0 was released and appear broken?

**Reply by alexg-k** [reply]
I am wondering the same. Moreover it seems my wife deleted some photos in her phone. Can I extract the images from immich somehow and the re-upload them, if I have not yet deleted any photos (though they are not visible in the web UI anymore)?

**Reply by jschwalbe** [reply]
@alextran1502 any ideas on this one?

**Reply by alextran1502** [reply]
@jschwalbe we don't have that mechanism to tell which one hasn't linked correctly. The only way to solve it is to see the live and motion part are shown as separated asset on the timeline, then you will delete them and reupload via the mobile app

**Reply by jschwalbe** [reply]
@alextran1502 thank you for the reply! Couple quick questions:
1. if I haven't uploaded anything to immich in a few months, am I safe?
2. is it any live photo ever uploaded or just after a certain version?
3. if i delete the files off immich, will the app automatically upload or must i specifically trigger a re-upload?

thanks!

**Reply by alextran1502** [reply]
@jschwalbe 
1. Yes
2. After a certain version where the bug was introduced probably 78 or 79
3. It will reupload when you initialize the upload process

**Comment by mpetereit** [comment]
`chmod: changing permissions of '/var/lib/postgresql/data': Operation not permitted
PostgreSQL Database directory appears to contain a database; Skipping initialization
postgres: could not access the server configuration file "/var/lib/postgresql/data/postgresql.conf": Permission denied`

After the Update the Database is not starting up anymore. Its only showing that the permission is wrong now for some reason.

**Reply by mpetereit** [reply]
Found my Problem:
Iam running immich on Windows with WSL2. The mounts come directly from a windows share and Databases dont really like this. Had the same problem with Nextcloud too. I created a VHD now just for the DB and copied all files in there and edited the Volume mountpoint. 
So just the DB runs now inside the VHD and the rest is still mapped from WSL2

**Comment by jho1670** [comment]
With my giant library, the containers restart/come online _so much faster now_. Thank you for this great improvement!

**Comment by dsm1212** [comment]
i upgraded from 1.91.1 to latest a few minutes ago and now i'm getting "Invalid configuration file" with no indication what is invalid. Was there some config removed in the last 48 hours?

**Reply by adambeck7** [reply]
I just updated and everything seems fine. Post your `docker-compose` and your `.env`. Tough to say what's wrong without seeing that.

**Comment by dsm1212** [comment]
Looking through the commits I found that clipEncoding changed name to smartSearch in the config file. I fixed this and restarted. Why did no one else hit this??

**Comment by MrMarble** [comment]
I have immich running in kubernetes and after the update the immich-server pod fails to start, there aren't any logs, it just hangs there until is restarted for not meeting the readiness probes.
I tried creating a new database and got the same problem, must be something with immich itself

**Reply by adambeck7** [reply]
Did you change the log level env? As well as removing the typesense containers? And have you tried `1.91.3` yet?

**Reply by MrMarble** [reply]
I'm currently using `1.91.3` with the same problem.
`LOG_LEVEL` was the default value `log`, tried with `verbose` and still same issue, I attached a shell into the container and removed the `bufferedLogs` option from nestFactory and was able to get some output but it was stuck there:
![imagen](https://github.com/immich-app/immich/assets/4268580/81b8933a-6991-4b46-9b8c-60ccb8bd8f8a)

I changed postgres image to `ghcr.io/bo0tzz/cnpgvecto.rs:14.10-v0.1.11`, created a new database for immich and installed vectors extension with `CREATE EXTENSION vectors CASCADE;` which outputs correctly
![imagen](https://github.com/immich-app/immich/assets/4268580/ca599bb3-a075-4c4e-bd74-b44400ad52bd)
removed typesense containers and started immich. Everything was working fine on version `1.90.2`

I don't see anything on postgres logs, like is not even connecting, here are the manifest I use to deploy immich
https://github.com/MrMarble/home-ops/blob/main/kubernetes/home-cluster/apps/default/immich/app/server/helmrelease.yaml

**Reply by jrasm91** [reply]
How long is the readiness probe? The database connection timeout is 10 seconds. If the readiness probe kills the container before the 10 second timeout is hit, you won't ever see the "Unable to connect to the database" error, which is what I'm thinking is happening.

**Reply by MrMarble** [reply]
> How long is the readiness probe? The database connection timeout is 10 seconds. If the readiness probe kills the container before the 10 second timeout is hit, you won't ever see the "Unable to connect to the database" error, which is what I'm thinking is happening.

The default is like 3 mins, I'll try removing the probe completely to see if that makes any difference

**Reply by MrMarble** [reply]
@jrasm91 I tried removing the probes and nothing, been running for 16 minutes
![imagen](https://github.com/immich-app/immich/assets/4268580/4d7ffa1e-a0fa-4862-9418-241b8d84a947)

**Comment by vnt87** [comment]
So when upgrading from 1.89.0, I was being an idiot and changed the database password for some reason. After getting a `Password authentication failed for user "postgres"` error I searched around and found #1145 and followed the advise to "resolve" it by removing the db persistent volume (did I mention I was an idiot?). 
Anyway I now end up with a brand new instance of Immich. 
Thankfully all the actually data is still there in my binded NAS folder, but with a blank db I don't have any photos, or album, or facial recognition data, or anything at all in my Immich instance. Any pointer on how I can repopulate the db?

**Reply by jrasm91** [reply]
Do you have a backup of the database....

**Reply by adambeck7** [reply]
Live and learn I guess. Here's the docs on making backups. They did a great job keeping it super simple https://immich.app/docs/administration/backup-and-restore/

You should probably just run that on a cron. If you did happen to make a backup, just run the restore process as documented above.

**Reply by dsm1212** [reply]
I used that link myself to upgrade postgres and it makes no mention of the need to reset your password when going to through 13-14. Fortunately I knew how to go into psql and reset it but maybe that's what the OP hit. I mentioned this issue earlier in this thread I think. There is a nice upgrade article here that mentions this issue:

https://www.ajfriesen.com/postgresql-upgrades-docker/

**Reply by vnt87** [reply]
I don't have a backup unfortunately :(. I was going to do it at some point but procrastination got the best of me. Oh well, like @adambeck7 said, live and learn I guess.
Now what would my options be at this point? My understanding is that there's no function within Immich that let it repopulate the db by scanning a folder structure, it's strictly upload only, right? In that case I assume my next action would be to reimport all the files using the CLI, but do I need to create a separate upload destination or I can just keep everything where they are?

**Reply by dsm1212** [reply]
Read up on external files. I don’t even use upload.

**Reply by adambeck7** [reply]
@vnt87 no, they allow external libraries now: https://immich.app/docs/features/libraries/
